gateway_code/open_nodes/node_pycom.py

```diff
@@ -80,7 +108,9 @@ def setup(self, firmware_path=None):
             ssh -L 20000:<pycom node>:20000 <login>@<site>.iot-lab.info
             socat PTY,link=/tmp/ttyS0,echo=0,crnl TCP:localhost:20000
         """
-        ret_val = gateway_code.common.wait_tty(self.TTY, LOGGER, timeout=10)
+        if self.expansion_board is True:
```

Comment:
Is the expansion board compatible with pytrack/pysense expansion boards ? (I think it is but unsure) Have you tried that ?

Reply:
Expansion bard, Pytrack and Pysense are all of them shields/daughter board. You can not plugged them together, they are exclusive. You can plugged Fipy/LoPy/Gpy on any of these shields.

Reply:
Sure. But my question was more about the compatibility of the initial code with the expansion board. Is it so different from Pytrack/Pysense that it needs a special case ?
What kind of board are you plugging on top the expansion board ? In Saclay, Fipy are deployed for example.

Reply:
Ok but it seems that there's an expansion board v3.0 that is compatible with pysense/pytrack. So it's more a matter of hardware version rather than a expansion board variant.
I understand the benefit of the v3.1 as you describe (can you share a good link where it's well described, I didn't find one, maybe add it to the node class docstring).

Reply:
I found [this page](https://docs.pycom.io/datasheets/expansionboards/expansion3/#differences-between-v20-and-v30) that explains the differences. Apparently the pysense/pytrack v2 are compatible to what you describe. The V3.1 doesn't provide super critical features compared to v3.0.

I'll check the versions of the expansion boards deployed in Saclay to see if they can be compatible with the update.py script.

Reply:
I've tested the Pycom `update.py` script with the different shields (EB, Pysense, Pytrack) used on IoT-LAB. Good news, they work finally all ! So I will update the PR with removing the if statement for the EB shield; and by default use the Pycom scripts for properly erase the filesystem during setup.

* expansion board 3.1 + LoPy4

```
$ python ./updater.py list
/dev/cu.usbmodemPyf25f471  [Expansion3 - Pycom] [USB VID:PID=04D8:EF98 SER=Pyf25f47 LOCATION=20-1.2.4]

python ./updater.py -p /dev/cu.usbmodemPyf25f471 erase_fs
Running in PIC mode
Flash file system erasure successful
```

* pysense v1.0 + LoPy4

```
$ python ./updater.py list
/dev/cu.usbmodemPy3434341  [Pysense - Pycom] [USB VID:PID=04D8:F012 SER=Py343434 LOCATION=20-1.2.4]


$ python ./updater.py -p /dev/cu.usbmodemPy3434341 erase_fs
Running in PIC mode
Flash file system erasure successful
```

* pytrack v1.0 + LoPy4

```
$ python ./updater.py list
/dev/cu.usbmodemPy3434341  [Pytrack - Pycom] [USB VID:PID=04D8:F013 SER=Py343434 LOCATION=20-1.2.4]

$ python ./updater.py -p /dev/cu.usbmodemPy3434341 erase_fs
Running in PIC mode
Flash file system erasure successful
```